### PR TITLE
Reimplement `UsState` as `UsSubdivision`

### PR DESCRIPTION
**Pull request by promag**
The provinces package is only used to query for US states. This PR removes the dependency in favour of having the list of 50 states.

Note: not considering the following (non state) divisions of US:

 - Outlying areas:
   - American Samoa (US-AS)
   - Guam (US-GU)
   - Northern Mariana Islands (US-MP)
   - Puerto Rico (US-PR)
   - United States Minor Outlying Islands (US-UM)
   - Virgin Islands, U.S. (US-VI)

 - Districts:
   - District of Columbia (US-DC)

R=@ruimarinho

### CODE REVIEW

**Comment by nunofgs** [comment]
@promag: ping

File: dist/asserts/us-state-assert.js [hunk 1]
```diff
@@ -7,15 +7,11 @@
```
Comment (ruimarinho) [inline dist/asserts/us-state-assert.js]: Can you please remove this dist file from the commit?

File: dist/asserts/us-state-assert.js [hunk 2]
```diff
@@ -39,7 +35,7 @@ exports['default'] = function () {
       throw new _validatorJs.Violation(_this, value, { value: _validatorJs.Validator.errorCode.must_be_a_string });
     }
 
-    if (_lodash.some(_provinces2['default'], { short: value, country: 'US' }) !== true) {
+    if (_.includes(states, value) === false) {
```
Comment (ruimarinho) [inline dist/asserts/us-state-assert.js]: `!_.includes(states, value)`?

File: src/asserts/us-state-assert.js [hunk 3]
```diff
@@ -28,7 +80,7 @@ export default function() {
       throw new Violation(this, value, { value: Validator.errorCode.must_be_a_string });
     }
 
-    if (some(provinces, { short: value, country: 'US' }) !== true) {
+    if (includes(states, value) === false) {
```
Comment (ruimarinho) [inline src/asserts/us-state-assert.js]: `!includes(states, value)`?

**Comment by ruimarinho** [comment]
As discussed offline, we should rename this to `UsSubdivision` assert and allow different types of subdivisions (states, outlying, federal) to be enabled or disabled as needed.

**Pull request merged**